test/frontend/test_template_tags.py

```diff
@@ -133,3 +134,28 @@ def test_socialaccount_providers_gitlab(self):
         social_providers = socialaccount_providers(context)
         self.assertEqual(1, len(social_providers.keys()))
         self.assertEqual(GitLabProvider, list(social_providers)[0].__class__)
+
+    def test_catch_error_when_status_missing(self):
+        # Test that if the status for a build gets deleted, that this is
+        # handled appropriately, rather than causing a crash.
+
+        # create the group, project and build
+        self.group = models.Group.objects.create(slug="mygroup")
+        self.project = self.group.projects.create(slug="myproject")
+        self.build = self.project.builds.create(version="mybuild")
+        self.project.latest_build = self.build
+
+        # Set build status to None
+        self.build.status = None
+
+        # Try to call project_status when status is None
+        missing_project_status_error = False
+        try:
+            status = project_status(self.project)
+        except models.Build.status.RelatedObjectDoesNotExist:
```

Comment:
Thank you, Charles - would there be a way to test the page loads successfully, rather than testing for the particular error? If that would be possible, would it be more useful?

Otherwise, I can just remove the test :+1:

Reply:
I think I understand what you're suggesting but I am having trouble getting the test to fail when I remove the bug fix.
I have tried things along the lines of:
```
    def test_project_list_no_status(self):
        self.build = self.project.builds.create(version="mybuild")
        self.project.latest_build = self.build
        self.project.latest_build = None
        self.hit('/mygroup/?all_projects=1')
```
Am I missing something here?

Reply:
Sorry for delay!

Your test didn't fail because you only changed the variable, not the database. Try this:
```python
    def test_project_list_no_status(self):
        self.build = self.project.builds.create(version="mybuild")
        self.build.status.delete() # this actually deletes the status object in db
        self.hit('/mygroup/?all_projects=1')
```

Also, I removed `self.project.latest_build` because this is set here: https://github.com/Linaro/squad/blob/ab8551115a7b60cd574f04723718df53d52f5c08/squad/frontend/views.py#L145

Does it make sense to you?

Reply:
Ah, great - thank you! I tried to use `delete()` originally but I must have got it slightly wrong as it seems to work now :)

Reply:
Have added the test and pushed - let me know if that looks ok :+1: